sycl/include/CL/sycl/handler.hpp

```diff
@@ -663,16 +663,51 @@ class handler {
 #endif
   }
 
-  // template <typename KernelName, typename WorkgroupFunctionType, int
-  // dimensions>
-  // void parallel_for_work_group(range<dimensions> numWorkGroups,
-  //                              WorkgroupFunctionType KernelFunc);
+  template <typename KernelName = csd::auto_name, typename KernelType, int Dims>
+  void parallel_for_work_group(range<Dims> NumWorkGroups,
+                               KernelType KernelFunc) {
+    using NameT = typename csd::get_kernel_name_t<KernelName, KernelType>::name;
+#ifdef __SYCL_DEVICE_ONLY__
+    kernel_parallel_for_work_group<NameT, KernelType, Dims>(KernelFunc);
+#else
+    MNDRDesc.setNumWorkGroups(NumWorkGroups);
+    StoreLambda<NameT, KernelType, Dims>(std::move(KernelFunc));
```

Comment:
Right. I just followed the pattern set by other invocation APIs here. Should we file an issue here and fix all at once?

Reply:
Yes, it is probably a more global issue and in the meantime you can skip the `std::move`.
The compile can figure out this `std::move` by itself anyway since it is the last use of the variable passed by copy, I think.